# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsPresContext.h
# Commit: f8ca02252dfc
# Full Hash: f8ca02252dfc48d7342703e6a635d510b9dc0b5d
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-04-02 09:51:45
# Regressor Bug: 1619349
# File Overlap Count: 1
# Description:
#   Bug 1619349 - patch 1 - Record statistics about font-matching behavior in the presContext. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D67787
# ==============================================================================

diff -r 01a2a527aa1a -r f8ca02252dfc layout/base/nsPresContext.h
--- a/layout/base/nsPresContext.h	Wed Apr 01 21:42:11 2020 +0000
+++ b/layout/base/nsPresContext.h	Wed Apr 01 21:42:36 2020 +0000
@@ -68,6 +68,7 @@
 class nsIWidget;
 class nsDeviceContext;
 class gfxMissingFontRecorder;
+struct FontMatchingStats;
 
 namespace mozilla {
 class AnimationEventDispatcher;
@@ -834,6 +835,7 @@
   }
 
   gfxTextPerfMetrics* GetTextPerfMetrics() { return mTextPerf.get(); }
+  FontMatchingStats* GetFontMatchingStats() { return mFontStats.get(); }
 
   bool IsDynamic() {
     return (mType == eContext_PageLayout || mType == eContext_Galley);
@@ -1194,6 +1196,8 @@
   // text performance metrics
   mozilla::UniquePtr<gfxTextPerfMetrics> mTextPerf;
 
+  mozilla::UniquePtr<FontMatchingStats> mFontStats;
+
   mozilla::UniquePtr<gfxMissingFontRecorder> mMissingFonts;
 
   nsRect mVisibleArea;